src/components/MapComponent.vue

```diff
@@ -0,0 +1,262 @@
+<template>
+  <div id="map-component">
+    <b-button class="label" type="submit" @click="showLabels">
+      {{`${(this.labelsVisible ? "Hide" : "Show")} Block Numbers`}}
+    </b-button>
+  </div>
+</template>
+
+<script>
+import { mapState } from 'vuex';
+import { loadModules } from 'esri-loader';
+import mapConstants from '../constants/mapConstants';
+
+export default {
+  name: 'mapComponent',
+  props: {
+    reservedFilter: {
+      type: Number,
+      required: false,
+    },
+
+    pushBlock: {
+      type: Function,
+      required: false,
+    },
+
+    activeBlockId: {
+      type: String,
+      required: false,
+    },
+
+    isAdminMap: {
+      type: Boolean,
+      required: false,
+    },
+  },
+
+  data: () => ({
+    labelsVisible: true,
+  }),
+
+  computed: {
+    ...mapState({
+      reservedBlocks: 'reservedBlocks',
+    }),
+
+    // Determines which blocks to display depending on the user is looking at
+    // reserve new blocks, current reservations, or admin map
+    blockFilter() {
+      if (this.isAdminMap) {
+        // Admins can see all current reserved blocks by any user
+        return 'RESERVED = 1 OR RESERVED = 2';
+      } if (this.reservedFilter === 0) {
+        // If reserving new blocks, only show blocks that are not reserved
+        return 'RESERVED = 0';
+      } if (this.reservedBlocks.length > 0) {
+        // If there are given IDs, have the map only show the given blocks
+        return `ID = ${this.reservedBlocks.join(' OR ID = ')} AND RESERVED = 1`;
+      }
+      // Will not display any blocks
+      return '1=0';
+    },
+
+    // The content of the popup when clicking on a block
+    // https://developers.arcgis.com/javascript/latest/api-reference/esri-PopupTemplate.html
+    blockPopupTemplate() {
+      return {
+        // autocasts as new PopupTemplate()
+        title: '{ID}', // Show attribute value
+        content: this.reservationToString,
+        outFields: ['*'],
+        actions: this.blockActions,
+      };
+    },
+
+    // Sets whether a user can reserve, release, or complete a block depending
+    // on which map screen they are viewing
+    blockActions() {
+      if (this.reservedFilter === 0) {
+        return [mapConstants.ADD_TO_RESERVE];
+      }
+      return [mapConstants.ADD_TO_RELEASE, mapConstants.ADD_TO_COMPLETE];
+    },
+
+    // contains the information about each of the blocks, each block is a feature.
+    // https://developers.arcgis.com/javascript/latest/api-reference/esri-layers-FeatureLayer.html
+    blockLayerObject() {
```

Comment:
Before, complete blocks and reserved blocks had separate renderers on the admin map. Now all blocks are displayed from this featureLayer